src/components/carousel.tsx

```diff
@@ -0,0 +1,51 @@
+"use client";
+
+import { Swiper, SwiperSlide } from "swiper/react";
+import "swiper/css";
+import "swiper/css/pagination";
+
+import { Autoplay, Pagination } from "swiper/modules";
+
+interface CarouselProps {
+  items: React.ReactNode[];
+  class?: string;
+}
+
+export default function Carousel({ items }: CarouselProps) {
+  const hasText = items.some((item) => item !== undefined);
+
+  return (
+    <Swiper
+      slidesPerView={1}
+      spaceBetween={30}
+      centeredSlides={true}
+      autoplay={{
+        delay: 2000,
```

Comment:
This delay should be received as a parameter and should be optional (no delay if the parameter isn't specified).